android/app/src/main/java/app/organicmaps/location/TrackRecordingService.java

```diff
@@ -163,7 +165,7 @@ public int onStartCommand(@NonNull Intent intent, int flags, int startId)
     {
       try
       {
-        startForeground(TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build());
+        ServiceCompat.startForeground(this, TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build(), ServiceInfo.FOREGROUND_SERVICE_TYPE_LOCATION);
```

Comment:
What is the point of specifying the service type that was defined in the manifest and already has a service type specified?
```xml
<service android:name=".location.TrackRecordingService"
      android:foregroundServiceType="location"
      android:exported="false"
      android:enabled="true"
      android:stopWithTask="false"
      />
````

If this is a legitimate change then I'd prefer to set this value: [FOREGROUND_SERVICE_TYPE_MANIFEST](https://developer.android.com/reference/android/content/pm/ServiceInfo.html#FOREGROUND_SERVICE_TYPE_MANIFEST). It should automatically use all values from manifest and no need to define them in two places.

Reply:
I am also not very sure that it gonna solve the issue correctly or not but i read that from android 14 above we need to provide the type of foreground service. Since i was unable to regenerate the issue i cannot say it will resolve it since everything else seems fine.

Reply:
https://developer.android.com/about/versions/14/changes/fgs-types-required#location

In this doc it states the steps while starting foreground service which requires location service

Kindly have a look @AndrewShkrob , i think this will probably solve the issue